### PR TITLE
get rid of some core-api mentions too

### PR DESCRIPTION
**Pull request by freben**

### CODE REVIEW

File: docs/getting-started/project-structure.md [hunk 1]
```diff
@@ -166,12 +166,12 @@ are separated out into their own folder, see further down.
   plugin and [techdocs-cli](https://github.com/backstage/techdocs-cli).
 
 - [`test-utils/`](https://github.com/backstage/backstage/tree/master/packages/test-utils) -
-  This package contains specific testing facilities used when testing
-  `core-api`.
+  This package contains more general purpose testing facilities for testing a
+  Backstage App or its plugins.
```
Comment (freben) [inline docs/getting-started/project-structure.md:170]: bit of a mixup between these two :)

File: packages/cli/src/commands/versions/lint.ts [hunk 2]
```diff
@@ -27,8 +27,8 @@ export const includedFilter = (name: string) =>
 
 // Packages that are not allowed to have any duplicates
 const FORBID_DUPLICATES = [
-  /^@backstage\/core$/,
-  /^@backstage\/core-api$/,
+  /^@backstage\/core-app-api$/,
+  /^@backstage\/core-plugin-api$/,
```
Comment (Rugvip) [inline packages/cli/src/commands/versions/lint.ts:31]: This one is allowed to be duplicated, so can drop it

File: packages/cli/src/commands/versions/lint.ts [hunk 3]
```diff
@@ -27,8 +27,8 @@ export const includedFilter = (name: string) =>
...
   /^@backstage\/plugin-/,
```
Comment (Rugvip) [inline packages/cli/src/commands/versions/lint.ts:32]: Duplicates of `@backstage/plugin-catalog-react` is allowed as well. Not really part of this change but if you're here anyway 😅

Comment (freben) [reply packages/cli/src/commands/versions/lint.ts:31]: Alright, added!

File: packages/cli/src/commands/versions/lint.ts [hunk 4]
```diff
@@ -27,13 +27,21 @@ export const includedFilter = (name: string) =>
...
-  /^@backstage\/core-api$/,
+  /^@backstage\/core-app-api$/,
   /^@backstage\/plugin-/,
 ];
 
+// There are some packages that ARE explicitly allowed to have duplicates since
+// they handle that appropriately. This takes precedence over FORBID_DUPLICATES
+// above.
+const ALLOW_DUPLICATES = [
+  /^@backstage\/core-plugin-api$/,
```
Comment (Rugvip) [inline packages/cli/src/commands/versions/lint.ts:38]: This doesn't get matched in the above, but I guess it's nice to have it explicitly here anyway? 🤷

**Review by Rugvip — APPROVED** [review]
🧹

**Review by Rugvip — COMMENTED** [review]
Oh changeset for the CLI change tho!

File: packages/cli/src/commands/versions/lint.ts (hunk 4 above)
Comment (freben) [reply packages/cli/src/commands/versions/lint.ts:38]: yeah i know, felt it was clearer / more intentional and safer against any impl change we might make elsewhere :)

**Comment by freben** [comment]
ahh yass changeset added now
